lib/server-info-route-handler.js

```diff
@@ -0,0 +1,150 @@
+const _ = require('lodash')
+const { calculateBounds } = require('@terraformer/spatial')
+const {
+  getCollectionCrs,
+  getGeometryTypeFromGeojson,
+  normalizeExtent,
+  normalizeSpatialReference
+} = require('./helpers')
+const { serverMetadata: serverMetadataDefaults } = require('./defaults')
+const debug = process.env.KOOP_LOG_LEVEL === 'debug' || process.env.LOG_LEVEL === 'debug'
+
+function serverMetadata (json, { query = {} } = {}) {
+  const { extent, metadata, ...rest } = json
+  const { maxRecordCount, hasStaticData, description } = { ...metadata, ...rest }
+  const spatialReference = getSpatialReference(json, query)
+  const { layers, tables } = normalizeInput(json)
+  const fullExtent = getServiceExtent({ extent, metadata, layers, spatialReference })
+
+  // TODO reproject default extents when non WGS84 CRS is found or passed
+
+  return _.defaults({
+    spatialReference,
+    fullExtent,
+    initialExtent: fullExtent,
+    layers: layers.map(layerInfo),
+    tables: tables.map(tableInfo),
+    serviceDescription: description,
+    maxRecordCount: maxRecordCount || _.get(layers, '[0].metadata.maxRecordCount'),
+    hasStaticData: typeof hasStaticData === 'boolean' ? hasStaticData : false
+  }, serverMetadataDefaults)
+}
+
+function normalizeInput (input) {
+  const { type, tables = [], layers = [] } = input
+
+  if (type === 'FeatureCollection') {
+    const geometryType = getGeometryTypeFromGeojson(input)
+    if (geometryType) return { layers: [input], tables }
+    return { tables: [input], layers }
+  }
```

Comment:
Why is the original `tables` or `layers` array overwritten?

Reply:
see previous comment.